apps/emqx_exhook/src/emqx_exhook_handler.erl

```diff
@@ -307,9 +307,8 @@ stringfy(Term) when is_atom(Term) ->
 stringfy(Term) ->
     unicode:characters_to_binary((io_lib:format("~0p", [Term]))).
 
-hexstr(<<>>) -> [];
-hexstr(<<H:4, L:4, B/binary>>) ->
-    iolist_to_binary([hexchar(H), hexchar(L)] ++ hexstr(B)).
+hexstr(B) ->
+    << <<(hexchar(H)), (hexchar(L))>> || <<H:4, L:4>> <= B>>.
 
 hexchar(0) -> $0;
```

Comment:
this could be done in a more efficient one-line code.
`hexchar(N) when N >=0 andalso N =< 15 -> N + $0.`

Reply:
I have checked it with `statistics/1` and it's more efficient indeed. Cool!

Reply:
d'oh. sorry, two lines of code.